mne/beamformer/_compute_beamformer.py

```diff
@@ -44,93 +43,67 @@ def _check_src_type(filters):
+    if noise_cov is None:
+        noise_cov = make_ad_hoc_cov(info, std=1.)
+    forward, info_picked, gain, _, orient_prior, _, trace_GRGT, noise_cov, \
+        whitener = _prepare_forward(
+            forward, info, noise_cov, 'auto', loose, rank=rank, pca=pca,
+            use_cps=True, exp=exp, limit_depth_chs=limit_depth_chs,
+            combine_xyz=combine_xyz, limit=limit,
+            allow_fixed_depth=allow_fixed_depth)
+    is_free_ori = not is_fixed_orient(forward)  # could have been changed
+    nn = forward['source_nn']
+    if is_free_ori:  # take Z coordinate
+        nn = nn[2::3]
+    nn = nn.copy()
+    vertno = _get_vertno(forward['src'])
+    if forward['surf_ori']:
+        nn[...] = [0, 0, 1]  # align to local +Z coordinate
+    if pick_ori is not None and not is_free_ori:
+        raise ValueError(
+            'Normal or max-power orientation (got %r) can only be picked when '
+            'a forward operator with free orientation is used.' % (pick_ori,))
     if pick_ori == 'normal' and not forward['surf_ori']:
-        # XXX eventually this could just call convert_forward_solution
         raise ValueError('Normal orientation can only be picked when a '
                          'forward operator oriented in surface coordinates is '
                          'used.')
-    if pick_ori == 'normal' and not forward['src'][0]['type'] == 'surf':
+    if pick_ori == 'normal' and not forward['src'].kind == 'surface':
         raise ValueError('Normal orientation can only be picked when a '
                          'forward operator with a surface-based source space '
                          'is used.')
-    # Check whether data and forward model have same compensation applied
-    _check_compensation_grade(forward['info'], info, 'forward')
-
-    # Restrict forward solution to selected channels
-    info_ch_names = [ch['ch_name'] for ch in info['chs']]
-    ch_names = [info_ch_names[k] for k in picks]
-    fwd_ch_names = forward['sol']['row_names']
-    # Keep channels in forward present in info:
-    fwd_ch_names = [ch for ch in fwd_ch_names if ch in ch_names]
-    forward = pick_channels_forward(forward, fwd_ch_names)
-    picks_forward = [fwd_ch_names.index(ch) for ch in ch_names]
-
-    # Get gain matrix (forward operator)
-    nn = forward['source_nn']
-    if is_free_ori:  # take Z coordinate
-        nn = nn[2::3]
-    nn = nn.copy()
-    if forward['surf_ori']:
-        nn[...] = [0, 0, 1]  # align to local +Z coordinate
-    if label is not None:
-        vertno, src_sel = label_src_vertno_sel(label, forward['src'])
-        nn = nn[src_sel]
-
-        if is_free_ori:
-            src_sel = 3 * src_sel
-            src_sel = np.c_[src_sel, src_sel + 1, src_sel + 2]
-            src_sel = src_sel.ravel()
+    del forward, info
 
-        G = forward['sol']['data'][:, src_sel]
+    # Undo the scaling that MNE prefers
+    scale = np.sqrt((noise_cov['eig'] > 0).sum() / trace_GRGT)
+    gain /= scale
```

Comment:
Remind me why the scaling is necessary? Doesn't this change the output?

Reply:
The opposite of this scaling is in the unified function, so this line just undoes it (to preserve the scaling currently in `master`)

Reply:
So it preserves the tiny beamformer units? :)